src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -5523,6 +5523,18 @@ cdef class Polynomial(CommutativePolynomial):
         if tgt.ngens() > 1 and tgt._has_singular:
             g = flatten(self).gcd(flatten(other))
             return flatten.section()(g)
+        try:
+            P = self.parent()
+            g = self._pari_with_name().gcd(other._pari_with_name())
```

Comment:
`pari(self)` is shorter, although this is technically internal code so it is supposedly fine… but apparently `pari(x)` is less likely to work as evidenced by

```
sage: R.<x> = Zmod(10^19)[]
sage: pari(x)
```